internal/controllers/clusterclass/clusterclass_controller.go

```diff
@@ -260,10 +269,20 @@ func (r *Reconciler) reconcileVariables(ctx context.Context, clusterClass *clust
 			"VariableDiscovery failed: %s", kerrors.NewAggregate(errs))
 		return errors.Wrapf(kerrors.NewAggregate(errs), "failed to discover variables for ClusterClass %s", clusterClass.Name)
 	}
-	clusterClass.Status.Variables = allVars
+
+	statusVarList := []clusterv1.ClusterClassStatusVariable{}
+	for _, variable := range allVariableDefinitions {
+		statusVarList = append(statusVarList, *variable)
+	}
+	// Alphabetically sort the variables by name. This ensures no unnecessary updates to the ClusterClass status.
```

Comment:
If I'm not wrong, if we want to achieve stable ordering we must sort also definitions nested in each entry; this can be done both here (preferred) or where we add definitions

Reply:
+1 It's basically the same as the order in which patches are applied is always stable because we always just iterate through the array in the ClusterClass.

Reply:
Maybe let's add a short note about that this part is already stable here?